applications/CompressiblePotentialFlowApplication/custom_conditions/incompressible_adjoint_potential_wall_condition.cpp

```diff
@@ -0,0 +1,261 @@
+    mpPrimalCondition->Set(Flags(*this));
+    mpPrimalCondition->InitializeSolutionStep(rCurrentProcessInfo);
+}
+template <class TPrimalCondition>
+void AdjointIncompressiblePotentialWallCondition<TPrimalCondition>::GetValuesVector(Vector& rValues, int Step) 
+{
+
+    KRATOS_TRY
+
+    if(rValues.size() != TNumNodes)
+        rValues.resize(TNumNodes, false);
+
+    bool is_kutta=false;
+    for(unsigned int i=0; i<TNumNodes; i++){
+        if (GetGeometry()[i].FastGetSolutionStepValue(DISTANCE)<0.0){
+            is_kutta=true;
+            break;
+        }
+    }
+    for(unsigned int i=0; i<TNumNodes; i++){
+        if(is_kutta){
+            if(GetGeometry()[i].FastGetSolutionStepValue(DISTANCE)<0.0)
+                rValues[i] = GetGeometry()[i].FastGetSolutionStepValue(ADJOINT_VELOCITY_POTENTIAL);
+            else
+                rValues[i] = GetGeometry()[i].FastGetSolutionStepValue(ADJOINT_AUXILIARY_VELOCITY_POTENTIAL);
+        }
+        else
+            rValues[i] = GetGeometry()[i].FastGetSolutionStepValue(ADJOINT_VELOCITY_POTENTIAL);
+    }
+
+    KRATOS_CATCH("");
+
+}
+
+template <class TPrimalCondition>
+void AdjointIncompressiblePotentialWallCondition<TPrimalCondition>::CalculateLeftHandSide(MatrixType &rLeftHandSideMatrix,
+                            ProcessInfo &rCurrentProcessInfo) 
+{
+    VectorType RHS;
+    this->CalculateLocalSystem(rLeftHandSideMatrix, RHS, rCurrentProcessInfo);
+    rLeftHandSideMatrix.clear();
+}
+
+template <class TPrimalCondition>
+void AdjointIncompressiblePotentialWallCondition<TPrimalCondition>::CalculateSensitivityMatrix(const Variable<array_1d<double,3> >& rDesignVariable,
+                                        Matrix& rOutput,
+                                        const ProcessInfo& rCurrentProcessInfo) 
+{
+    if (rOutput.size1() != TNumNodes)
+        rOutput.resize(TDim*TNumNodes, TNumNodes, false);
+    rOutput.clear();
+}
+
+template <class TPrimalCondition>
+void AdjointIncompressiblePotentialWallCondition<TPrimalCondition>::CalculateLocalSystem(MatrixType &rLeftHandSideMatrix,
+                            VectorType &rRightHandSideVector,
+                            ProcessInfo &rCurrentProcessInfo) 
+{               
+    if (rLeftHandSideMatrix.size1() != TNumNodes)
+        rLeftHandSideMatrix.resize(TNumNodes, TNumNodes, false);
+    if (rRightHandSideVector.size() != TNumNodes)
+        rRightHandSideVector.resize(TNumNodes, false);
+    rLeftHandSideMatrix.clear();
+}
+
+/// Check that all data required by this condition is available and reasonable
+template <class TPrimalCondition>
+int AdjointIncompressiblePotentialWallCondition<TPrimalCondition>::Check(const ProcessInfo& rCurrentProcessInfo) 
+{
+    KRATOS_TRY;
+
+    int Check = Condition::Check(rCurrentProcessInfo); // Checks id > 0 and area > 0
+
+    if (Check != 0)
+    {
+        return Check;
+    }
+    else
+    {
+        // Check that all required variables have been registered
```

Comment:
according to @jcotela these checks are no longer needed => the app would not compile if these variables would be removed